docs/reference/contributing/storage/storage.md

```diff
@@ -1,3 +1,27 @@
 <h2 id="contributing-storage">Storage</h2>
 
-The two storage APIs are BlockDevice and FileSystem. These APIs provide filesystem and file operations on a block device, as well as an interface for access to block-based storage.
+Storage support is split between file systems and their underlying block device support. The [storage API page](/docs/v5.6/reference/storage-overview.html) has more information on existing APIs in Mbed OS for both interfaces.
+
+#### Block Device
+
+Backing file systems on new hardware requires adding a block device implementation. You can extend the [BlockDevice](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/bd/HeapBlockDevice.h) class to provide support for unsupported storage.
+
+If you want to port a new file system to Mbed OS on existing storage options, you can skip to the following section.
```

Comment:
Query: What do you mean by the following section? File systems two lines below? Or the next page? What should the reader do if the reader doesn't want to port a new file system to Mbed OS?

Reply:
Correct, the following section is referring to the file system section. 

A reader going through this section will want one of the following 3 situations:

1. Add a new physical or virtual storage device, requiring the base abilities of reading, writing, and erasing it.
2. Add a new filesystem implementation on an existing storage solution (Say on a SD card which Mbed OS supports). They do not need to port any  block device code themselves (what the last line in the Block Device section is referring to).
3. Combine the above two, a reader may want to implement a new filesystem solution on a block device that is not supported in Mbed OS.